### PR TITLE
chore: bump dependency versions

### PR DESCRIPTION
**Pull request by bradzacher**
## Overview

This just does a bump of all our dependencies to pin their versions to the latest in all packages except the website packages.

The remaining outdated packages are intentionally outdated ones

image

- nx deps to be upgraded later (cc @JamesHenry)
  - @nrwl/jest      
  - @nrwl/nx-cloud  
  - @nrwl/workspace 
  - @nx/jest        
  - @nx/linter      
  - @nx/workspace
  - nx
- prettier has a major to work through
- `@types/node` - TBH we should probably downgrade this to v16 - given that's the lowest version we support. For now I just left it at the highest v18
- `ts-node` because of the bug (we need to finish migrating off this to tsx tbh)

### CODE REVIEW

**Comment by JamesHenry** [comment]
Did you use nx migrate for this?

**Comment by bradzacher** [comment]
> Did you use nx migrate for this?

@JamesHenry I haven't bumped `nx` in this PR - I figured we could do that separately given that might need nx config changes too.

**Comment by JamesHenry** [comment]
D’oh sorry I misread your description - yeah we always wanna use Nx migrate for that reason. I can do that in a follow up later

File: packages/ast-spec/src/declaration/TSImportEqualsDeclaration/fixtures/external-module-ref-string/snapshots/5-AST-Alignment-AST.shot [hunk 1]
```diff
@@ -52,8 +52,7 @@ exports[`AST Fixtures declaration TSImportEqualsDeclaration external-module-ref-
         },
       },
     ],
--   sourceType: 'module',
-+   sourceType: 'script',
```
Comment (JoshuaKGoldberg) [inline packages/ast-spec/src/declaration/TSImportEqualsDeclaration/fixtures/external-module-ref-string/snapshots/5-AST-Alignment-AST.shot:56]: ...cool.

File: packages/repo-tools/src/postinstall.mts [hunk 2]
```diff
@@ -1,5 +1,12 @@
-import * as execa from 'execa';
+import { $ as $_config } from 'execa';
```
Comment (JoshuaKGoldberg) [inline packages/repo-tools/src/postinstall.mts:1]: Nit: I might call this `$config`...

File: patches/@microsoft+api-extractor+7.36.1.patch [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
+index 31b46f8..e7081c6 100644
 --- a/node_modules/@microsoft/api-extractor/lib/api/ExtractorConfig.js
 +++ b/node_modules/@microsoft/api-extractor/lib/api/ExtractorConfig.js
-@@ -669,5 +669,5 @@ ExtractorConfig.FILENAME = 'api-extractor.json';
+@@ -668,6 +668,6 @@ ExtractorConfig.FILENAME = 'api-extractor.json';
   */
  ExtractorConfig._tsdocBaseFilePath = path.resolve(__dirname, '../../extends/tsdoc-base.json');
  ExtractorConfig._defaultConfig = node_core_library_1.JsonFile.load(path.join(__dirname, '../schemas/api-extractor-defaults.json'));
 -ExtractorConfig._declarationFileExtensionRegExp = /\.d\.ts$/i;
 +ExtractorConfig._declarationFileExtensionRegExp = /\.d\.(c|m)?ts$/i;
```
Comment (JoshuaKGoldberg) [inline patches/@microsoft+api-extractor+7.36.1.patch:10]: Just bumped https://github.com/microsoft/rushstack/pull/4023#issuecomment-1631687085

**Review by JoshuaKGoldberg — APPROVED** [review]
Cartoony/drawn ship choppily animated to be bouncing on some waves

**Pull request merged**